qa/setup_packages.py

```diff
@@ -62,67 +69,79 @@ def get_pyvers_name(name, cuda):
              pass
     return ""
 
+def get_version(package_version_entry):
+    if isinstance(package_version_entry, tuple):
+        return package_version_entry[0]
+    elif isinstance(package_version_entry, str):
+        return package_version_entry
+    else:
+        return "Default"
+
 def print_configs(cuda):
-    for key in packages.keys():
+    for key in package_data.keys():
         print (key + ":")
-        for val in get_package(packages, key, cuda):
-            if val == None:
-                val = "Default"
-            elif val.startswith('http'):
-                val = get_pyvers_name(val, cuda)
-            print ('\t' + val)
+        for val in get_package_list(package_data, key, cuda):
+            print ('\t' + get_version(val))
 
 def get_install_string(variant, use, cuda):
     ret = []
-    for key in packages.keys():
+    for key in package_data.keys():
         if key not in use:
             continue
-        tmp = variant % len(get_package(packages, key, cuda))
-        val = get_package(packages, key, cuda)[tmp]
-        if val == None:
-            ret.append(key)
-        elif val.startswith('http'):
-            ret.append(get_pyvers_name(val, cuda))
-        else:
-            ret.append(key + "==" + val)
-        variant = variant // len(get_package(packages, key, cuda))
+        pkg_list_len = len(get_package_list(package_data, key, cuda))
+        idx = variant % pkg_list_len
+        val = get_package_list(package_data, key, cuda)[idx]
+        pkg_str = key
+        if isinstance(val, str):
+            if val.startswith('http'):
```

Comment:
Do we need that case anymore?

Reply:
We don't but I was hesitant to delete it just in case it turns out that we need it any time soon. Removing it wouldn't affect anything though